spec/consensus/light-client.md

```diff
@@ -60,259 +61,368 @@ For the purpose of this lite client specification, we assume that the Tendermint
 
-- the lite client must retry to retrieve correct headers from another full node
-  - it picks a new full node
-  - it restarts _Bisection_
-  - there might be optimizations; a lite client may not need to call _Commit(k)_, for a height _k_ for which it already has a signed header it trusts.
-  - how to make sure that a lite client can communicate with a correct full node will be the focus of a separate document (recall Issue 3 from "Context of this document").
+We consider the following set-up:
+- the lite client communicates with one full node
+- the lite client locally stores all the headers that has passed basic verification and that are within lite client trust period. In the pseudo code below we write *Store(header)* for this. If a header failed to verify, then
+the full node we are talking to is faulty and we should disconnect from it and reinitialise lite client.
+- If *CanTrust* returns *error*, then the lite client has seen a forged header or the trusted header has expired (it is outside its trusted period).
+  * In case of forged header, the full node is faulty so lite client should disconnect and reinitialise with new trusted header. If the trusted header has expired,
+  we need to reinitialise lite client with new trusted header (that is within its trusted period), but we don't necessarily need to disconnect from the full node
+  we are talking to (as we haven't observed full node misbehavior in this case).
+
+**Auxiliary Functions.** We will use the  function ```votingpower_in(V1,V2)``` to compute the voting power the validators in set V1 have according to their voting power in set V2;
+we will write ```totalVotingPower(V)``` for ```votingpower_in(V,V)```, which returns the total voting power in V.
+We further use the function ```signers(Commit)``` that returns the set of validators that signed the Commit.
+
+**CheckSupport.** The following function defines skipping condition under the Tendermint Failure model, i.e., it defines when we can trust the header h2 based on header h1.
+Time validity of a header is captured by the ```isWithinTrustedPeriodWithin``` function that depends on lite client trusted period (`LITE_CLIENT_TRUSTED_PERIOD`) and it returns
+true in case the header is within its lite client trusted period.
+```verify``` function is capturing basic header verification, i.e., it ensures that the header is signed by more than 2/3 of the voting power of the corresponding validator set.
 
-**Auxiliary Functions.** We will use the function `votingpower_in(V1,V2)` to compute the voting power the validators in set V1 have according to their voting power in set V2;
-we will write `totalVotingPower(V)` for `votingpower_in(V,V)`, which returns the total voting power in V.
-We further use the function `signers(Commit)` that returns the set of validators that signed the Commit.
+```go
+  // return true if header is within its lite client trusted period; otherwise it returns false
+  func isWithinTrustedPeriod(h) bool {
```

Comment:
Actually this is very important because it clarifies that there may be redundant calls to `isWithinTrustedPeriod` in `VerifyHeader`, unless that function has access to the system clock! So we need to clarify what we expect to be given the time from outside and what has access to fetch the current time from the clock.